platform/web/display_server_web.cpp

```diff
@@ -576,9 +576,17 @@ void DisplayServerWeb::_mouse_update_mode() {
 
 void DisplayServerWeb::mouse_set_mode(MouseMode p_mode) {
 	ERR_FAIL_INDEX(p_mode, MouseMode::MOUSE_MODE_MAX);
-	if (p_mode == mouse_mode_base) {
+
+	if (mouse_mode_override_enabled) {
+		mouse_mode_base = p_mode;
+		// No need to update, as override is enabled.
+		return;
+	}
+	if (p_mode == mouse_mode_base && p_mode == mouse_get_mode()) {
```

Comment:
`mouse_get_mode()` is specific to the Web platform, as we don't store a cached version of the state. We query the browser directly.

Reply:
> It also fixes an issue on the Web platform where the mouse mode would refuse to capture the mouse twice.

Other platform seems to automatically recapture mouse if you refocus the app. For example, Windows call mode update from window activation handlers to restore it, maybe Web should store it and auto-recapture mouse instead of requiring to set in the second time?

Reply:
@bruvzg Not possible. Because the browser is letting you a way to escape from websites, by pressing escape. And capturing the mouse is only possible from a direct action from the user: i.e. a click. So it's not possible to capture the mouse without intent from the user. If the user escaped, we cannot capture until a new click.

See https://developer.mozilla.org/en-US/docs/Web/API/Element/requestPointerLock#security for more info.